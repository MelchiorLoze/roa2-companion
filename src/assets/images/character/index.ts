import { type ImageSource } from 'expo-image';

export const AbsaIcon: ImageSource = require('./absa.png');
export const ClairenIcon: ImageSource = require('./clairen.png');
export const EtalusIcon: ImageSource = require('./etalus.png');
export const ForsburnIcon: ImageSource = require('./forsburn.png');
export const FleetIcon: ImageSource = require('./fleet.png');
export const KraggIcon: ImageSource = require('./kragg.png');
export const LoxodontIcon: ImageSource = require('./loxodont.png');
export const MaypulIcon: ImageSource = require('./maypul.png');
export const OlympiaIcon: ImageSource = require('./olympia.png');
export const OrcaneIcon: ImageSource = require('./orcane.png');
export const RannoIcon: ImageSource = require('./ranno.png');
export const WrastorIcon: ImageSource = require('./wrastor.png');
export const ZetterburnIcon: ImageSource = require('./zetterburn.png');
