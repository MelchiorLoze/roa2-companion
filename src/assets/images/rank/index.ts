import type { ImageSource } from 'expo-image';

export const StoneIcon: ImageSource = require('./stone.png');
export const BronzeIcon: ImageSource = require('./bronze.png');
export const SilverIcon: ImageSource = require('./silver.png');
export const GoldIcon: ImageSource = require('./gold.png');
export const PlatinumIcon: ImageSource = require('./platinum.png');
export const DiamondIcon: ImageSource = require('./diamond.png');
export const MasterIcon: ImageSource = require('./master.png');
export const GrandmasterIcon: ImageSource = require('./grandmaster.png');
export const AethereanIcon: ImageSource = require('./aetherean.png');
