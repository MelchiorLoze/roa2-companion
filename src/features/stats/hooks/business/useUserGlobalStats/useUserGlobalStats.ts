import { type PlayerStatistics, StatisticName } from '../../../types/stats';
import { useGetPlayerStatistics } from '../../data/useGetPlayerStatistics/useGetPlayerStatistics';

const getGlobalStats = (rawStats: PlayerStatistics) => {
  const gameCount = rawStats[StatisticName.TOTAL_SESSIONS_PLAYED] ?? 0;
  const winCount = rawStats[StatisticName.BETA_WINS] ?? 0;
  const winRate = gameCount ? (winCount / gameCount) * 100 : 0;

  return { gameStats: { gameCount, winCount, winRate } } as const;
};

export const useUserGlobalStats = () => {
  const { statistics: rawStats, refetch, isLoading, isRefetching } = useGetPlayerStatistics();

  return {
    stats: rawStats ? getGlobalStats(rawStats) : undefined,
    refresh: () => void refetch(),
    isLoading,
    isRefreshing: isRefetching,
  } as const;
};
