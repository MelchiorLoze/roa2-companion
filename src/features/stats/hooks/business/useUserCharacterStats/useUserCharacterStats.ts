import { Character } from '@/types/character';

import { type PlayerStatistics, StatisticName, type UserData } from '../../../types/stats';
import { useGetPlayerStatistics } from '../../data/useGetPlayerStatistics/useGetPlayerStatistics';
import { useGetUserReadOnlyData } from '../../data/useGetUserReadOnlyData/useGetUserReadOnlyData';

const getCharacterStats = (rawStats: PlayerStatistics, userData: UserData) =>
  Object.values(Character).map((character) => ({
    character,
    gameCount: rawStats[StatisticName[`${character.toUpperCase()}_MATCH_COUNT` as keyof typeof StatisticName]] ?? 0,
    level: userData.characterData[character]?.lvl ?? 0,
  }));

export const useUserCharacterStats = () => {
  const {
    statistics: rawStats,
    refetch: refetchPlayerStatistics,
    isLoading: isLoadingRawStats,
  } = useGetPlayerStatistics();
  const { userData, refetch: refetchUserData, isLoading: isLoadingUserData } = useGetUserReadOnlyData();

  const refresh = () => {
    void refetchPlayerStatistics();
    void refetchUserData();
  };

  const isLoading = isLoadingRawStats || isLoadingUserData;

  const stats = rawStats && userData && !isLoading ? getCharacterStats(rawStats, userData) : undefined;

  return { stats, refresh, isLoading } as const;
};
