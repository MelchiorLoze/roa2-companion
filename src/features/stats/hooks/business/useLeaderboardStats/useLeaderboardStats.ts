import { type barDataItem, type lineDataItem } from 'react-native-gifted-charts';
import { useUnistyles } from 'react-native-unistyles';
import { type UnistylesTheme } from 'react-native-unistyles/lib/typescript/src/types';

import { useSeason } from '@/features/stats/contexts/SeasonContext/SeasonContext';
import { getRank, type LeaderboardEntry, MAX_AETHEREAN_PLAYERS, type Rank } from '@/types/rank';

import { useCommunityLeaderboard } from '../../data/useCommunityLeaderboard/useCommunityLeaderboard';

const DISTRIBUTION_PRECISION = 10;

const roundedElo = (elo: number) => Math.floor(elo / DISTRIBUTION_PRECISION) * DISTRIBUTION_PRECISION;

type RankDistributionBarDataItem = barDataItem;
const getRankDistribution = (
  leaderboardEntries: LeaderboardEntry[],
  theme: UnistylesTheme,
): RankDistributionBarDataItem[] => {
  const result = leaderboardEntries.reduce(
    (acc, entry) => {
      const rank = getRank(entry.elo, entry.position);
      if (acc[rank]?.value) acc[rank].value++;
      else acc[rank] = { value: 1, frontColor: theme.color[rank] };
      return acc;
    },
    {} as Record<Rank, RankDistributionBarDataItem>,
  );

  return Object.values(result);
};

type EloDistributionLineDataItem = { elo: number } & lineDataItem;
const getEloDistribution = (
  leaderboardEntries: LeaderboardEntry[],
  firstPlayerElo: number,
  lastPlayerElo: number,
  userElo: number,
): EloDistributionLineDataItem[] => {
  const totalEloRange = firstPlayerElo - lastPlayerElo + 1;
  const result = Array.from(
    { length: Math.ceil(totalEloRange / DISTRIBUTION_PRECISION) },
    (_, i) => i * DISTRIBUTION_PRECISION + roundedElo(lastPlayerElo),
  ).reduce(
    (acc, elo) => ({ ...acc, [elo]: { elo: elo, value: 0 } as EloDistributionLineDataItem }),
    {} as Record<number, EloDistributionLineDataItem>,
  );

  leaderboardEntries.forEach((entry) => {
    const elo = roundedElo(entry.elo);
    if (result[elo].value) result[elo].value++;
    else result[elo].value = 1;
  });

  const userEloRounded = roundedElo(userElo);
  if (result[userEloRounded]) result[userEloRounded].dataPointColor = 'red';

  return Object.values(result).sort((a, b) => a.elo - b.elo);
};

export const useLeaderboardStats = (userElo = 0) => {
  const { leaderboardId } = useSeason();
  const { leaderboardEntries, isLoading: isLoadingLeaderboard } = useCommunityLeaderboard(leaderboardId);

  const { theme } = useUnistyles();

  const firstPlayerElo = leaderboardEntries[0]?.elo ?? 0;
  const lastPlayerElo = Math.min(0, leaderboardEntries[leaderboardEntries.length - 1]?.elo ?? 0);
  const lastAethereanElo =
    leaderboardEntries.findLast((entry) => entry.elo > 1800 && entry.position < MAX_AETHEREAN_PLAYERS)?.elo ?? 0;

  return {
    firstPlayerElo,
    lastPlayerElo,
    lastAethereanElo,
    rankDistribution: getRankDistribution(leaderboardEntries, theme),
    eloDistribution: getEloDistribution(leaderboardEntries, firstPlayerElo, lastPlayerElo, userElo),
    isLoading: isLoadingLeaderboard,
  };
};
