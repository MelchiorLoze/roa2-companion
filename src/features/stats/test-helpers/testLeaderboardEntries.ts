import { type LeaderboardEntry } from '../types/rank';

export const testLeaderboardEntries: LeaderboardEntry[] = [
  // Aetherean
  { steamId: 1, position: 1, elo: 2162 },
  { steamId: 2, position: 2, elo: 2089 },
  { steamId: 3, position: 3, elo: 1910 },
  { steamId: 4, position: 4, elo: 1837 },
  // Grandmaster
  { steamId: 5, position: 5, elo: 1791 },
  // Master
  { steamId: 6, position: 6, elo: 1655 },
  { steamId: 7, position: 7, elo: 1649 },
  { steamId: 8, position: 8, elo: 1523 },
  // Diamond
  { steamId: 9, position: 9, elo: 1450 },
  { steamId: 10, position: 10, elo: 1300 },
  // Platinum
  { steamId: 11, position: 11, elo: 1258 },
  { steamId: 12, position: 12, elo: 1111 },
  // Gold
  { steamId: 13, position: 13, elo: 1000 },
  { steamId: 14, position: 14, elo: 920 },
  // Silver
  { steamId: 15, position: 15, elo: 820 },
  { steamId: 16, position: 16, elo: 752 },
  // Bronze
  { steamId: 17, position: 17, elo: 699 },
  { steamId: 18, position: 18, elo: 510 },
  // Stone
  { steamId: 19, position: 19, elo: 400 },
  { steamId: 20, position: 20, elo: 300 },
  { steamId: 25, position: 25, elo: -100 },
];
