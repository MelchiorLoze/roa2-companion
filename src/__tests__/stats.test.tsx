import { render } from '@testing-library/react-native';

import Stats from '@/app/(private)/stats';
import { useLeaderboardStats } from '@/features/stats/hooks/business/useLeaderboardStats/useLeaderboardStats';
import { useUserCharacterStats } from '@/features/stats/hooks/business/useUserCharacterStats/useUserCharacterStats';
import { useUserGlobalStats } from '@/features/stats/hooks/business/useUserGlobalStats/useUserGlobalStats';
import { useUserRankedStats } from '@/features/stats/hooks/business/useUserRankedStats/useUserRankedStats';
import { testLeaderboardEntries } from '@/features/stats/test-helpers/testLeaderboardEntries';
import { Rank } from '@/features/stats/types/rank';
import { Character } from '@/types/character';

jest.mock('@/features/stats/contexts/SeasonContext/SeasonContext', () => ({
  useSeason: jest.fn().mockReturnValue({
    season: {
      index: 1,
      name: 'Season 1',
      isFirst: true,
      isLast: false,
    },
    leaderboardId: 789,
    isLoading: false,
    setPreviousSeason: jest.fn(),
    setNextSeason: jest.fn(),
  }),
}));

jest.mock('@/features/stats/hooks/business/useUserRankedStats/useUserRankedStats');
const useUserRankedStatsMock = jest.mocked(useUserRankedStats);
useUserRankedStatsMock.mockReturnValue({
  stats: {
    elo: 925,
    position: 123,
    rank: Rank.GOLD,
    setStats: { setCount: 100, winCount: 75, winRate: 75 },
  },
  refresh: jest.fn(),
  isLoading: false,
});

jest.mock('@/features/stats/hooks/business/useUserGlobalStats/useUserGlobalStats');
const useUserGlobalStatsMock = jest.mocked(useUserGlobalStats);
useUserGlobalStatsMock.mockReturnValue({
  stats: {
    gameStats: { gameCount: 500, winCount: 300, winRate: 60 },
  },
  refresh: jest.fn(),
  isLoading: false,
});

jest.mock('@/features/stats/hooks/business/useUserCharacterStats/useUserCharacterStats');
const useUserCharacterStatsMock = jest.mocked(useUserCharacterStats);
useUserCharacterStatsMock.mockReturnValue({
  stats: [
    { character: Character.KRAGG, gameCount: 20, level: 3 },
    { character: Character.CLAIREN, gameCount: 50, level: 5 },
    { character: Character.OLYMPIA, gameCount: 10, level: 10 },
    { character: Character.RANNO, gameCount: 30, level: 5 },
  ],
  refresh: jest.fn(),
  isLoading: false,
});

jest.mock('@/features/stats/hooks/business/useLeaderboardStats/useLeaderboardStats');
const useLeaderboardStatsMock = jest.mocked(useLeaderboardStats);
useLeaderboardStatsMock.mockReturnValue({
  firstPlayerElo: 2162,
  lastPlayerElo: -100,
  lastAethereanElo: 1837,
  leaderboardEntries: testLeaderboardEntries,
  isLoading: false,
});

describe('Stats', () => {
  it('matches the snapshot', () => {
    const tree = render(<Stats />).toJSON();

    // Remove circular references for snapshot testing
    // eslint-disable-next-line @typescript-eslint/no-unsafe-member-access
    delete tree.props.refreshControl;

    expect(tree).toMatchSnapshot();
  });
});
