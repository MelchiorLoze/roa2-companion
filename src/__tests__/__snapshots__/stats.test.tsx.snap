// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Stats matches the snapshot 1`] = `
<RCTScrollView
  contentContainerStyle={
    {
      "gap": 16,
      "padding": 16,
    }
  }
>
  <RCTRefreshControl />
  <View>
    <ViewManagerAdapter_ExpoLinearGradient
      colors={
        [
          4283136696,
          0,
        ]
      }
      endPoint={
        [
          1,
          0,
        ]
      }
      startPoint={
        [
          0.3333333333333333,
          0,
        ]
      }
      style={
        {
          "gap": 8,
          "padding": 8,
        }
      }
    >
      <View
        style={
          {
            "flexDirection": "row",
            "justifyContent": "space-between",
          }
        }
      >
        <Text
          style={
            {
              "color": "white",
              "fontFamily": "AgencyFB-Bold",
              "fontSize": 20,
              "textTransform": "uppercase",
            }
          }
        >
          Ranked - 
          Season 1
        </Text>
        <View
          style={
            {
              "alignItems": "center",
              "flexDirection": "row",
            }
          }
        >
          <View
            accessibilityState={
              {
                "busy": undefined,
                "checked": undefined,
                "disabled": true,
                "expanded": undefined,
                "selected": undefined,
              }
            }
            accessibilityValue={
              {
                "max": undefined,
                "min": undefined,
                "now": undefined,
                "text": undefined,
              }
            }
            accessible={true}
            collapsable={false}
            focusable={true}
            onBlur={[Function]}
            onClick={[Function]}
            onFocus={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            role="button"
            style={
              [
                {
                  "alignItems": "center",
                  "backgroundColor": "transparent",
                  "justifyContent": "center",
                },
                {
                  "padding": 2,
                },
              ]
            }
          >
            <
              color="#AAAAAA"
              name="arrow-back"
              size={24}
            />
          </View>
          <Text
            style={
              {
                "color": "white",
                "fontFamily": "AgencyFB-Bold",
                "fontSize": 20,
                "textTransform": "uppercase",
              }
            }
          >
            S
            1
          </Text>
          <View
            accessibilityState={
              {
                "busy": undefined,
                "checked": undefined,
                "disabled": false,
                "expanded": undefined,
                "selected": undefined,
              }
            }
            accessibilityValue={
              {
                "max": undefined,
                "min": undefined,
                "now": undefined,
                "text": undefined,
              }
            }
            accessible={true}
            collapsable={false}
            focusable={true}
            onBlur={[Function]}
            onClick={[Function]}
            onFocus={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            role="button"
            style={
              [
                {
                  "alignItems": "center",
                  "backgroundColor": "transparent",
                  "justifyContent": "center",
                },
                {
                  "padding": 2,
                },
              ]
            }
          >
            <
              color="white"
              name="arrow-forward"
              size={24}
            />
          </View>
        </View>
      </View>
      <View
        style={
          {
            "alignItems": "flex-end",
            "flexDirection": "row",
            "justifyContent": "space-between",
          }
        }
      >
        <View>
          <Text
            style={
              {
                "color": "white",
                "fontFamily": "FranklinGothicDemiCond-Regular",
                "fontSize": 16,
                "textTransform": "uppercase",
              }
            }
          >
            100
             sets: 
            75
             W -
             
            25
             L
          </Text>
          <Text
            style={
              {
                "color": "white",
                "fontFamily": "FranklinGothicDemiCond-Regular",
                "fontSize": 16,
                "textTransform": "uppercase",
              }
            }
          >
            Winrate: 
            75.00
            %
          </Text>
        </View>
        <View
          style={
            {
              "alignItems": "center",
              "flex": 1,
              "flexDirection": "row",
              "gap": 4,
              "justifyContent": "flex-end",
            }
          }
        >
          <ViewManagerAdapter_ExpoImage
            containerViewRef={"[React.ref]"}
            contentFit="contain"
            contentPosition={
              {
                "left": "50%",
                "top": "50%",
              }
            }
            height={24}
            nativeViewRef={"[React.ref]"}
            onError={[Function]}
            onLoad={[Function]}
            onLoadStart={[Function]}
            onProgress={[Function]}
            placeholder={[]}
            source={
              [
                {
                  "testUri": "../../../src/features/stats/assets/images/rank/gold.png",
                },
              ]
            }
            style={
              {
                "height": 24,
                "width": 24,
              }
            }
            transition={null}
            width={24}
          />
          <Text
            style={
              {
                "color": "white",
                "fontFamily": "FranklinGothicDemiCond-Regular",
                "fontSize": 16,
                "textTransform": "uppercase",
              }
            }
          >
            <Text
              style={
                {
                  "color": "#FFC64E",
                }
              }
            >
              925
            </Text>
             - #
            123
          </Text>
        </View>
      </View>
      <View>
        <View
          testID="rank-distribution"
        >
          <
            data={
              [
                {
                  "barWidth": 186.1246133451171,
                  "frontColor": "#7F7A66",
                  "topLabelComponent": [Function],
                  "value": 3,
                },
                {
                  "barWidth": 62.0415377817057,
                  "frontColor": "#B87A2E",
                  "topLabelComponent": [Function],
                  "value": 2,
                },
                {
                  "barWidth": 62.0415377817057,
                  "frontColor": "#C8C8D5",
                  "topLabelComponent": [Function],
                  "value": 2,
                },
                {
                  "barWidth": 62.0415377817057,
                  "frontColor": "#FFC64E",
                  "topLabelComponent": [Function],
                  "value": 2,
                },
                {
                  "barWidth": 62.0415377817057,
                  "frontColor": "#C3C4FF",
                  "topLabelComponent": [Function],
                  "value": 3,
                },
                {
                  "barWidth": 62.0415377817057,
                  "frontColor": "#95DFEC",
                  "topLabelComponent": [Function],
                  "value": 1,
                },
                {
                  "barWidth": 62.0415377817057,
                  "frontColor": "#9CE7AC",
                  "topLabelComponent": [Function],
                  "value": 3,
                },
                {
                  "barWidth": 42.80866106937693,
                  "frontColor": "#FC7575",
                  "topLabelComponent": [Function],
                  "value": 1,
                },
                {
                  "barWidth": 100.81749889527177,
                  "frontColor": "#BD7CC5",
                  "topLabelComponent": [Function],
                  "value": 4,
                },
              ]
            }
            disablePress={true}
            disableScroll={true}
            height={702}
            hideAxesAndRules={true}
            spacing={0}
            width={702}
            xAxisLabelsHeight={0}
            yAxisLabelWidth={0}
          />
        </View>
        <View
          style={
            {
              "position": "absolute",
            }
          }
          testID="elo-distribution"
        >
          <
            adjustToWidth={true}
            data={
              [
                {
                  "value": 0,
                },
                {
                  "value": 0,
                },
                {
                  "value": 0,
                },
                {
                  "value": 0,
                },
                {
                  "value": 0,
                },
                {
                  "value": 0,
                },
                {
                  "value": 0,
                },
                {
                  "value": 0,
                },
                {
                  "value": 0,
                },
                {
                  "value": 0,
                },
                {
                  "value": 0,
                },
                {
                  "value": 0,
                },
                {
                  "value": 0,
                },
                {
                  "value": 0,
                },
                {
                  "value": 0,
                },
                {
                  "value": 1,
                },
                {
                  "value": 0,
                },
                {
                  "value": 0,
                },
                {
                  "value": 0,
                },
                {
                  "value": 0,
                },
                {
                  "value": 1,
                },
                {
                  "value": 0,
                },
                {
                  "value": 0,
                },
                {
                  "value": 0,
                },
                {
                  "value": 0,
                },
                {
                  "value": 1,
                },
                {
                  "value": 0,
                },
                {
                  "value": 0,
                },
                {
                  "value": 0,
                },
                {
                  "value": 0,
                },
                {
                  "value": 0,
                },
                {
                  "value": 0,
                },
                {
                  "value": 0,
                },
                {
                  "value": 0,
                },
                {
                  "value": 1,
                },
                {
                  "value": 0,
                },
                {
                  "value": 0,
                },
                {
                  "value": 1,
                },
                {
                  "value": 0,
                },
                {
                  "value": 0,
                },
                {
                  "value": 0,
                },
                {
                  "value": 1,
                },
                {
                  "value": 0,
                },
                {
                  "value": 0,
                },
                {
                  "value": 0,
                },
                {
                  "value": 0,
                },
                {
                  "dataPointColor": "red",
                  "dataPointRadius": 2,
                  "value": 1,
                },
                {
                  "value": 0,
                },
                {
                  "value": 0,
                },
                {
                  "value": 0,
                },
                {
                  "value": 1,
                },
                {
                  "value": 0,
                },
                {
                  "value": 0,
                },
                {
                  "value": 0,
                },
                {
                  "value": 0,
                },
                {
                  "value": 1,
                },
                {
                  "value": 0,
                },
                {
                  "value": 0,
                },
                {
                  "value": 0,
                },
                {
                  "value": 0,
                },
                {
                  "value": 0,
                },
                {
                  "value": 0,
                },
                {
                  "value": 1,
                },
                {
                  "value": 0,
                },
                {
                  "value": 0,
                },
                {
                  "value": 1,
                },
                {
                  "value": 0,
                },
                {
                  "value": 0,
                },
                {
                  "value": 0,
                },
                {
                  "value": 0,
                },
                {
                  "value": 0,
                },
                {
                  "value": 0,
                },
                {
                  "value": 1,
                },
                {
                  "value": 0,
                },
                {
                  "value": 0,
                },
                {
                  "value": 0,
                },
                {
                  "value": 1,
                },
                {
                  "value": 0,
                },
                {
                  "value": 0,
                },
                {
                  "value": 0,
                },
                {
                  "value": 0,
                },
                {
                  "value": 0,
                },
                {
                  "value": 2,
                },
                {
                  "value": 0,
                },
                {
                  "value": 0,
                },
                {
                  "value": 0,
                },
                {
                  "value": 0,
                },
                {
                  "value": 0,
                },
                {
                  "value": 0,
                },
                {
                  "value": 1,
                },
                {
                  "value": 0,
                },
                {
                  "value": 1,
                },
                {
                  "value": 0,
                },
                {
                  "value": 0,
                },
                {
                  "value": 0,
                },
                {
                  "value": 1,
                },
                {
                  "value": 0,
                },
                {
                  "value": 0,
                },
                {
                  "value": 0,
                },
                {
                  "value": 0,
                },
                {
                  "value": 0,
                },
                {
                  "value": 0,
                },
                {
                  "value": 0,
                },
                {
                  "value": 0,
                },
                {
                  "value": 1,
                },
                {
                  "value": 0,
                },
                {
                  "value": 0,
                },
                {
                  "value": 0,
                },
                {
                  "value": 1,
                },
                {
                  "value": 1,
                },
                {
                  "value": 0,
                },
                {
                  "value": 0,
                },
                {
                  "value": 0,
                },
                {
                  "value": 0,
                },
              ]
            }
            dataPointsColor="transparent"
            disableScroll={true}
            height={702}
            hideAxesAndRules={true}
            initialSpacing={0}
            thickness={1}
            width={702}
            xAxisLabelsHeight={0}
            yAxisLabelWidth={0}
          />
        </View>
      </View>
    </ViewManagerAdapter_ExpoLinearGradient>
    <ViewManagerAdapter_ExpoLinearGradient
      colors={
        [
          4283136696,
          0,
        ]
      }
      endPoint={
        [
          1,
          0,
        ]
      }
      startPoint={
        [
          0.3333333333333333,
          0,
        ]
      }
      style={
        {
          "gap": 8,
          "padding": 8,
        }
      }
    >
      <Text
        style={
          {
            "color": "white",
            "fontFamily": "AgencyFB-Bold",
            "fontSize": 20,
            "textTransform": "uppercase",
          }
        }
      >
        Global
      </Text>
      <View>
        <Text
          style={
            {
              "color": "white",
              "fontFamily": "FranklinGothicDemiCond-Regular",
              "fontSize": 16,
              "textTransform": "uppercase",
            }
          }
        >
          500
           games: 
          300
           W -
           
          200
           L
        </Text>
        <Text
          style={
            {
              "color": "white",
              "fontFamily": "FranklinGothicDemiCond-Regular",
              "fontSize": 16,
              "textTransform": "uppercase",
            }
          }
        >
          Winrate: 
          60.00
          %
        </Text>
      </View>
    </ViewManagerAdapter_ExpoLinearGradient>
    <ViewManagerAdapter_ExpoLinearGradient
      colors={
        [
          4283136696,
          0,
        ]
      }
      endPoint={
        [
          1,
          0,
        ]
      }
      startPoint={
        [
          0.3333333333333333,
          0,
        ]
      }
      style={
        {
          "gap": 8,
          "padding": 8,
        }
      }
    >
      <Text
        style={
          {
            "color": "white",
            "fontFamily": "AgencyFB-Bold",
            "fontSize": 20,
            "textTransform": "uppercase",
          }
        }
      >
        Characters
      </Text>
      <View>
        <View
          style={
            {
              "borderBottomWidth": 1,
              "borderColor": "#FEDD84",
              "flexDirection": "row",
              "width": "100%",
            }
          }
        >
          <View
            style={
              {
                "alignItems": "center",
                "flex": 0.2,
                "justifyContent": "center",
              }
            }
          />
          <Text
            style={
              [
                {
                  "color": "white",
                  "fontFamily": "FranklinGothicDemiCond-Regular",
                  "fontSize": 16,
                  "textTransform": "uppercase",
                },
                {
                  "borderColor": "#FEDD84",
                  "borderLeftWidth": 1,
                  "flex": 0.4,
                  "paddingLeft": 8,
                },
              ]
            }
          >
            Level
          </Text>
          <Text
            style={
              [
                {
                  "color": "white",
                  "fontFamily": "FranklinGothicDemiCond-Regular",
                  "fontSize": 16,
                  "textTransform": "uppercase",
                },
                {
                  "borderColor": "#FEDD84",
                  "borderLeftWidth": 1,
                  "flex": 0.4,
                  "paddingLeft": 8,
                },
              ]
            }
          >
            Games
          </Text>
        </View>
        <View
          style={
            {
              "borderBottomWidth": 1,
              "borderColor": "#FEDD84",
              "flexDirection": "row",
              "width": "100%",
            }
          }
        >
          <View
            style={
              {
                "alignItems": "center",
                "flex": 0.2,
                "justifyContent": "center",
              }
            }
          >
            <ViewManagerAdapter_ExpoImage
              containerViewRef={"[React.ref]"}
              contentFit="contain"
              contentPosition={
                {
                  "left": "50%",
                  "top": "50%",
                }
              }
              height={24}
              nativeViewRef={"[React.ref]"}
              onError={[Function]}
              onLoad={[Function]}
              onLoadStart={[Function]}
              onProgress={[Function]}
              placeholder={[]}
              source={
                [
                  {
                    "testUri": "../../../src/assets/images/character/olympia.png",
                  },
                ]
              }
              style={
                {
                  "height": 24,
                  "width": 24,
                }
              }
              transition={null}
              width={24}
            />
          </View>
          <Text
            style={
              [
                {
                  "color": "white",
                  "fontFamily": "FranklinGothicDemiCond-Regular",
                  "fontSize": 16,
                  "textTransform": "uppercase",
                },
                {
                  "borderColor": "#FEDD84",
                  "borderLeftWidth": 1,
                  "flex": 0.4,
                  "paddingLeft": 8,
                },
              ]
            }
          >
            10
          </Text>
          <Text
            style={
              [
                {
                  "color": "white",
                  "fontFamily": "FranklinGothicDemiCond-Regular",
                  "fontSize": 16,
                  "textTransform": "uppercase",
                },
                {
                  "borderColor": "#FEDD84",
                  "borderLeftWidth": 1,
                  "flex": 0.4,
                  "paddingLeft": 8,
                },
              ]
            }
          >
            10
          </Text>
        </View>
        <View
          style={
            {
              "borderBottomWidth": 1,
              "borderColor": "#FEDD84",
              "flexDirection": "row",
              "width": "100%",
            }
          }
        >
          <View
            style={
              {
                "alignItems": "center",
                "flex": 0.2,
                "justifyContent": "center",
              }
            }
          >
            <ViewManagerAdapter_ExpoImage
              containerViewRef={"[React.ref]"}
              contentFit="contain"
              contentPosition={
                {
                  "left": "50%",
                  "top": "50%",
                }
              }
              height={24}
              nativeViewRef={"[React.ref]"}
              onError={[Function]}
              onLoad={[Function]}
              onLoadStart={[Function]}
              onProgress={[Function]}
              placeholder={[]}
              source={
                [
                  {
                    "testUri": "../../../src/assets/images/character/clairen.png",
                  },
                ]
              }
              style={
                {
                  "height": 24,
                  "width": 24,
                }
              }
              transition={null}
              width={24}
            />
          </View>
          <Text
            style={
              [
                {
                  "color": "white",
                  "fontFamily": "FranklinGothicDemiCond-Regular",
                  "fontSize": 16,
                  "textTransform": "uppercase",
                },
                {
                  "borderColor": "#FEDD84",
                  "borderLeftWidth": 1,
                  "flex": 0.4,
                  "paddingLeft": 8,
                },
              ]
            }
          >
            5
          </Text>
          <Text
            style={
              [
                {
                  "color": "white",
                  "fontFamily": "FranklinGothicDemiCond-Regular",
                  "fontSize": 16,
                  "textTransform": "uppercase",
                },
                {
                  "borderColor": "#FEDD84",
                  "borderLeftWidth": 1,
                  "flex": 0.4,
                  "paddingLeft": 8,
                },
              ]
            }
          >
            50
          </Text>
        </View>
        <View
          style={
            {
              "borderBottomWidth": 1,
              "borderColor": "#FEDD84",
              "flexDirection": "row",
              "width": "100%",
            }
          }
        >
          <View
            style={
              {
                "alignItems": "center",
                "flex": 0.2,
                "justifyContent": "center",
              }
            }
          >
            <ViewManagerAdapter_ExpoImage
              containerViewRef={"[React.ref]"}
              contentFit="contain"
              contentPosition={
                {
                  "left": "50%",
                  "top": "50%",
                }
              }
              height={24}
              nativeViewRef={"[React.ref]"}
              onError={[Function]}
              onLoad={[Function]}
              onLoadStart={[Function]}
              onProgress={[Function]}
              placeholder={[]}
              source={
                [
                  {
                    "testUri": "../../../src/assets/images/character/ranno.png",
                  },
                ]
              }
              style={
                {
                  "height": 24,
                  "width": 24,
                }
              }
              transition={null}
              width={24}
            />
          </View>
          <Text
            style={
              [
                {
                  "color": "white",
                  "fontFamily": "FranklinGothicDemiCond-Regular",
                  "fontSize": 16,
                  "textTransform": "uppercase",
                },
                {
                  "borderColor": "#FEDD84",
                  "borderLeftWidth": 1,
                  "flex": 0.4,
                  "paddingLeft": 8,
                },
              ]
            }
          >
            5
          </Text>
          <Text
            style={
              [
                {
                  "color": "white",
                  "fontFamily": "FranklinGothicDemiCond-Regular",
                  "fontSize": 16,
                  "textTransform": "uppercase",
                },
                {
                  "borderColor": "#FEDD84",
                  "borderLeftWidth": 1,
                  "flex": 0.4,
                  "paddingLeft": 8,
                },
              ]
            }
          >
            30
          </Text>
        </View>
        <View
          style={
            {
              "borderBottomWidth": 0,
              "borderColor": "#FEDD84",
              "flexDirection": "row",
              "width": "100%",
            }
          }
        >
          <View
            style={
              {
                "alignItems": "center",
                "flex": 0.2,
                "justifyContent": "center",
              }
            }
          >
            <ViewManagerAdapter_ExpoImage
              containerViewRef={"[React.ref]"}
              contentFit="contain"
              contentPosition={
                {
                  "left": "50%",
                  "top": "50%",
                }
              }
              height={24}
              nativeViewRef={"[React.ref]"}
              onError={[Function]}
              onLoad={[Function]}
              onLoadStart={[Function]}
              onProgress={[Function]}
              placeholder={[]}
              source={
                [
                  {
                    "testUri": "../../../src/assets/images/character/kragg.png",
                  },
                ]
              }
              style={
                {
                  "height": 24,
                  "width": 24,
                }
              }
              transition={null}
              width={24}
            />
          </View>
          <Text
            style={
              [
                {
                  "color": "white",
                  "fontFamily": "FranklinGothicDemiCond-Regular",
                  "fontSize": 16,
                  "textTransform": "uppercase",
                },
                {
                  "borderColor": "#FEDD84",
                  "borderLeftWidth": 1,
                  "flex": 0.4,
                  "paddingLeft": 8,
                },
              ]
            }
          >
            3
          </Text>
          <Text
            style={
              [
                {
                  "color": "white",
                  "fontFamily": "FranklinGothicDemiCond-Regular",
                  "fontSize": 16,
                  "textTransform": "uppercase",
                },
                {
                  "borderColor": "#FEDD84",
                  "borderLeftWidth": 1,
                  "flex": 0.4,
                  "paddingLeft": 8,
                },
              ]
            }
          >
            20
          </Text>
        </View>
      </View>
    </ViewManagerAdapter_ExpoLinearGradient>
  </View>
</RCTScrollView>
`;
